src/theory/arith/theory_arith_private.cpp

```diff
@@ -1900,7 +1908,14 @@ void TheoryArithPrivate::outputConflicts(){
         Debug("arith::conflict") << "(normalized to) " << conflict << endl;
       }
 
-      outputConflict(conflict);
+      if (options::proofNew())
```

Comment:
Can you eliminate use of checking proofNew?  Using a check that the proof node manager is non-null?